policy-controller/k8s/status/src/index.rs

```diff
@@ -125,21 +181,49 @@ impl Controller {
                     // any resources.
                     if self.leader {
                         if id.gkn.group == k8s::policy::HttpRoute::group(&()) {
-                            let api = k8s::Api::<k8s::policy::HttpRoute>::namespaced(self.client.clone(), &id.namespace);
-                            if let Err(error) = api.patch_status(&id.gkn.name, &patch_params, &patch).await {
-                                tracing::error!(namespace = %id.namespace, route = ?id.gkn, %error, "Failed to patch HTTPRoute");
-                            }
+                            self.patch_status::<k8s::policy::HttpRoute>(&id.gkn.name, &id.namespace, patch).await;
                         } else if id.gkn.group == k8s_gateway_api::HttpRoute::group(&()) {
-                            let api = k8s::Api::<k8s_gateway_api::HttpRoute>::namespaced(self.client.clone(), &id.namespace);
-                            if let Err(error) = api.patch_status(&id.gkn.name, &patch_params, &patch).await {
-                                tracing::error!(namespace = %id.namespace, route = ?id.gkn, %error, "Failed to patch HTTPRoute");
-                            }
+                            self.patch_status::<k8s_gateway_api::HttpRoute>(&id.gkn.name, &id.namespace, patch).await;
                         }
                     }
                 }
             }
         }
     }
+
+    async fn patch_status<K>(
+        &self,
+        name: &str,
+        namespace: &str,
+        patch: k8s::Patch<serde_json::Value>,
+    ) where
+        K: Resource<Scope = NamespaceResourceScope>,
+        <K as Resource>::DynamicType: Default,
+        K: DeserializeOwned,
+    {
+        let patch_params = k8s::PatchParams::apply(POLICY_API_GROUP);
+        let api = k8s::Api::<K>::namespaced(self.client.clone(), namespace);
+        let start = time::Instant::now();
+        match time::timeout(PATCH_TIMEOUT, api.patch_status(name, &patch_params, &patch)).await {
+            Ok(Ok(_)) => {
+                self.metrics.patch_succeeded.inc();
+                self.metrics
+                    .patch_duration
+                    .observe(start.elapsed().as_secs_f64());
+            }
+            Ok(Err(error)) => {
+                self.metrics.patch_failed.inc();
+                self.metrics
+                    .patch_duration
+                    .observe(start.elapsed().as_secs_f64());
+                tracing::error!(%namespace, %name, %error, "failed to patch HTTPRoute");
+            }
+            Err(_) => {
+                self.metrics.patch_timeout.inc();
+                tracing::error!(%namespace, %name, "timed out patching HTTPRoute");
```

Comment:
```suggestion
                tracing::error!(%namespace, %name, kind = %K::kind(&Default::default()), %error, "Patch failed");
            }
            Err(_) => {
                self.metrics.patch_timeout.inc();
                tracing::error!(%namespace, %name, kind = %K::kind(&Default::default()), "Patch timed out");
```

Even though this is only used with HTTPRoute now this will fit better more generally.

Also tiny nit: We generally (albeit inconsistently) prefer to start log messages with a Capital letter to differentiate them from contextual information.